crates/node/types/src/lib.rs

```diff
@@ -24,10 +24,16 @@ use reth_trie_db::StateCommitment;
 pub trait NodePrimitives {
     /// Block primitive.
     type Block;
+    /// Signed version of the transaction type.
+    type SignedTx;
+    /// Transaction type.
+    type Transaction;
```

Comment:
should those be integrated into a `Block` trait instead, because for block we also need those, right?

Reply:
think it makes sense to have these parallel to block too, so we can ref them without the trait bound `Block` on `NodePrimitives::Block`.